### PR TITLE
Scopes cookie

### PR DESCRIPTION
**Pull request by ajacksified**
:eyeglasses: @curioussavage or @umbrae 

This makes it so that if we change oauth scopes in the future, we can invalidate existing login cookies. (Like https://github.com/reddit/reddit-mobile/commit/73c9b73d2b8b36f060e9b844cbf33472e6474374)

### CODE REVIEW

**Comment by curioussavage** [comment]
:fish:

File: src/server.jsx [hunk 1]
```diff
@@ -341,16 +341,16 @@ class Server {
...
         yield next;
         return;
       }
 
-      if (skipAuth(app, this.url)) {
+      var now = new Date();
+      var expires = this.cookies.get('tokenExpires');
+
+      if (this.cookies.get('tokenScopes') !== app.oauthScopes) {
+        app.nukeTokens(this);
```
Comment (umbrae) [inline src/server.jsx]: Q: Does this mean that if we ever alter scopes for the app that every user will be logged out and forced to reauthenticate, or does it mean that we'll nuke the cookies and then regenerate a new token silently?

**Comment by umbrae** [comment]
:nail_care: for a Q

File: src/server.jsx (hunk 1 above)
Comment (ajacksified) [inline src/server.jsx]: Yes, they will be forced to reauthenticate - oauth flows don't allow for users to silently 'accept' new scopes, which would defeat a primary security model for oauth.

**Comment by umbrae** [comment]
Jack and I talked about this, and we came up with a middle ground:

1. Use this for now.
2. Create a ticket to make a special scope, "all", that we will use for mobile web, which will allow for all functionality for first party clients.
3. Once we have that built, we can update this to only use the "all" scope and not worry about forward compatibility.

:fish: on that

**Comment by kjoconnor** [comment]
We have that scope, it's `*`

**Comment by umbrae** [comment]
Yay! I thought we might have it, but I hadn't known for sure. @ajacksified can you test that?

Thank you @kjoconnor !

**Comment by kjoconnor** [comment]
I tried poking around in the code originally to see how we grant access to requesting that scope, but couldn't find it easily. I'm hoping it already uses one of the ways we whitelist clients, but if not I can check into it a little more deeply.

**Pull request merged**